modules/lib/lib-grid/src/main/resources/lib/xp/grid.js

```diff
@@ -0,0 +1,110 @@
+/* global __ */
+
+function requireNotNull(value, parameterName) {
+    if (typeof value === 'undefined' || value === null) {
+        throw `Parameter "${parameterName}" is required`;
+    }
+    return value;
+}
+
+function convertValue(value) {
+    if (Array.isArray(value)) {
+        return __.toScriptValue(value).getList();
+    } else if (typeof value === 'object') {
+        return __.toScriptValue(value).getMap();
+    } else {
+        return value;
+    }
+}
+
+/**
+ * Creates an instance of Shared Memory (Shared Map) which are shared across all applications and even cluster nodes.
+ *
+ * WARNING: Due to distributed nature of the Shared Map not all types of keys and values can be used. Strings, numbers, and pure JSON objects are supported. There is no runtime check for type compatibility due to performance reasons. The developer is also responsible for not modifying shared objects (keys and values) in place.
+ *
+ * @param mapId - map identifier
+ * @constructor
+ */
+function SharedMemory(mapId) {
+    var bean = __.newBean('com.enonic.xp.lib.grid.MemoryGridHandler');
+    this.map = bean.getMap(mapId);
+};
```

Comment:
Codacy has a fix for the issue: [Unnecessary semicolon.](https://app.codacy.com/gh/enonic/xp/pullRequest?prid=9567561)
```suggestion
}
```